lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -160,16 +160,13 @@ struct BoolCastOpConversion : public OpConversionPattern<BoolCastOp> {
   LogicalResult
   matchAndRewrite(BoolCastOp op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
-    if (cast<UnpackedType>(op.getInput().getType())
-            .castToSimpleBitVectorOrNull()) {
-      Type resultType = typeConverter->convertType(op.getInput().getType());
+    Type resultType = typeConverter->convertType(op.getInput().getType());
+    if (isa<IntegerType>(resultType)) {
```

Comment:
I think this can fail if `resultType` is the null Type, should be `isa_or_null`, right?

Reply:
Oh yeah good point. Let me add that.